src/watcher/monitor.py

```
@@ -105,6 +117,32 @@ def start(self) -> None:
+                and sys.platform != "win32"
+            ):
+                try:
+                    from utils.safedir import SafeDir
+
+                    watch_root = Path(self.config.watch_directories[0]).resolve()
+                    self.handler._safe_dir = SafeDir.open_root(watch_root)
+                    self.handler._watch_root = watch_root
+                    logger.debug(
+                        "FileMonitor: (re)initialized SafeDir for watch root %s", watch_root
+                    )
+                except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)